src/Health/HealthDAO.ts

```diff
@@ -105,4 +119,74 @@ export class HealthDAO {
       },
       { upsert: true },
     )
+  }
+
+  readonly isNameExisting: isNameExisting = async name => {
+    const existing = await this.collection.findOne({ name })
+    return !!existing
+  }
+
+  readonly insertIPFSFailure: insertIPFSFailure = async ({ failureReason, failureType, ipfsFileHash, failureTime }) => {
+    await this.collection.updateOne(
+      { name: 'ipfsDownloadRetries' },
+      {
+        $set: {
+          ipfsDownloadRetries: [
+            { ipfsFileHash, failureReason, failureType, lastDownloadAttemptTime: failureTime, downloadAttempts: 1 },
```

Comment:
Storing statistics is still the plan

Reply:
@kennylavender @lautarodragan Ok I was thinking that I would need to store the IPFS hashes still so that we could remove them whenever they were downloaded successfully and that we would only display the statistics. I've updated it so that it now only increases the 'hardFailures' and 'softFailures' counts based on the failureType. So the only thing it is storing now is a count of the 'HARD' IPFS Failures and a count of the 'SOFT' IPFS Failures.